consensus/merkle_proof/src/lib.rs

```diff
@@ -382,20 +382,19 @@ pub fn verify_merkle_proof(
 pub fn merkle_root_from_branch(leaf: H256, branch: &[H256], depth: usize, index: usize) -> H256 {
     assert_eq!(branch.len(), depth, "proof length should equal depth");
```

Comment:
I skipped it, but its possible to add another invariant here where the number of bits in the `index` should be more than the depth.

This should never trigger unless we are for example on a 16 bit architecture and the tree depth is something like 32/64. Though I imagine that others things assume usize is u32/u64, so they would break well before we reach this line